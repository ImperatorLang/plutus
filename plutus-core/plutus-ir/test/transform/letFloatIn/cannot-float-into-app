(let
  (nonrec)
  (termbind
    (strict)
    (vardecl addInteger (fun (con integer) (fun (con integer) (con integer))))
    (builtin addInteger)
  )
  (termbind
    (strict)
    (vardecl addInteger (fun (con integer) (fun (con integer) (con integer))))
    (lam x (con integer) (lam y (con integer) [ [ addInteger x ] y ]))
  )
  (termbind
    (strict)
    (vardecl f (fun (con integer) (con integer)))
    (lam y (con integer) [ [ addInteger y ] (con integer 3) ])
  )
  (termbind
    (nonstrict)
    (vardecl x (con integer))
    [ [ addInteger (con integer 1) ] (con integer 2) ]
  )
  [ [ addInteger [ f x ] ] [ f x ] ]
)
-- Both the function and the argument of the application mention `x`,
-- so `x` cannot be floated in.
