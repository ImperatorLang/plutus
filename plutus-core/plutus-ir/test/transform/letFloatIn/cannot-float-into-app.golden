(let
  (nonrec)
  (termbind
    (strict)
    (vardecl addInteger (fun (con integer) (fun (con integer) (con integer))))
    (builtin addInteger)
  )
  (termbind
    (strict)
    (vardecl addInteger (fun (con integer) (fun (con integer) (con integer))))
    (lam x (con integer) (lam y (con integer) [ [ addInteger x ] y ]))
  )
  (termbind
    (nonstrict)
    (vardecl x (con integer))
    [ [ addInteger (con integer 1) ] (con integer 2) ]
  )
  (termbind
    (strict)
    (vardecl f (fun (con integer) (con integer)))
    (lam y (con integer) [ [ addInteger y ] (con integer 3) ])
  )
  [ [ addInteger [ f x ] ] [ f x ] ]
)